### ISSUE TITLE
Archlinux pacman installation guidelines

### ISSUE THREAD

**Issue by soyuka** [emersion/xdg-desktop-portal-wlr#49]
Hi,

I'm having an issue where trying to install xdg-desktop-portal-wlr using something like:

```
(master ✓)❯ yay -S xdg-desktop-portal xdg-desktop-portal-wlr
resolving dependencies...
:: There are 2 providers available for xdg-desktop-portal-impl:
:: Repository extra
   1) xdg-desktop-portal-gtk  2) xdg-desktop-portal-kde

Enter a number (default=1):
```

Even when directly using `makepkg` in the AUR package xdg-desktop-portal-wlr-git `xdg-desktop-portal-impl` doesn't look to be guessed correctly:

```
(master ✓)❯ makepkg -si
==> Making package: xdg-desktop-portal-wlr-git v0.1.0.r1.ga105e0e-1 (Sat 05 Sep 2020 01:36:10 PM CEST)
==> Checking runtime dependencies...
==> Installing missing dependencies...
resolving dependencies...
:: There are 2 providers available for xdg-desktop-portal-impl:
:: Repository extra
   1) xdg-desktop-portal-gtk  2) xdg-desktop-portal-kde
```
Obviously I want neither of these. I'm able to install xdg-desktop-portal-wlr manually. Is there an easy fix to this ?

**Comment by danshick** [comment]
I have a guess, but I'm not certain about this.

If I understand correctly, yay invokes pacman as a wrapper for non-AUR packages, of which xdg-desktop-portal is one. When pacman attempts to resolve xdg-desktop-portal-impl, it only looks at implementations in the main repositories (both -gtk and -kde are in "extra"). Basically, pacman can't be made aware of -wlr because it won't ever look in AUR.

This is an implementation detail of your AUR helper, and not one that can be resolved through configuration.

If you tell yay to install only -wlr, I think it will resolve everything correctly as it will start with the AUR package and makepkg will resolve xdg-desktop-portal instead of pacman. I think it is clever enough to realize that the AUR package satisfies the -impl dependency.

Unless and until xdpw makes it into extra, I doubt this is fixable.

**Comment by soyuka** [comment]
Thanks for the answer, indeed I don't really care about the package manager (aurman, yay etc) and I would like to install these by using makepkg if I can. 

In fact, the whole point of installing these via makepkg would be to use https://gitlab.gnome.org/feaneron/obs-xdg-portal that has a dependency through xdg-desktop-portal (via pkgconfig). 

```
... -S xdg-desktop-portal-wlr
```

Doens't work either as it looks for xdg-desktop-portal that itself looks for `*-impl` without checking the AUR.

May I ask how you would install these tools? Or maybe what would you recommend as installation method that "just works"?

**Comment by emersion** [comment]
(Closing because this is an Arch issue.)

**Comment by soyuka** [comment]
TLDR if you have the issue, just install `xdg-desktop-portal-kde`

**Comment by danshick** [comment]
xdg-desktop-portal-wlr is now in the Arch "community" repository, so just install with pacman now unless you have a compelling reason to use the AUR -git version.
